test/constrain_reef_clusters.jl

```diff
@@ -0,0 +1,77 @@
+using Test
+using ADRIA: decision.constrain_reef_cluster
+using LinearAlgebra
+using Random
+using Statistics
+
+@testset "Constrain spatial groups" begin
+    n_sites = 30
+    site_ids = collect(1:n_sites)
+    area_to_seed = 695.11
+    orig_site_order = shuffle(Vector(1:n_sites))
+    site_order = copy(orig_site_order)
+    available_space = rand(Uniform(area_to_seed + 100.0, area_to_seed + 1000.0), 30)
+    n_site_iv = 5
+
+    prefsites = site_order[1:5]
+    reef_locs = [fill("1", 10)..., fill("2", 10)..., fill("3", 10)...]
+
+    s_order = Union{Float64,Int64}[Int64.(site_order) rand(n_sites)]
+    # All selected sites are in the same reef, so 2 should be replaced
+    reef_locs[prefsites] .= "2"
+    # Empty ranking just for testing
+    rankings = Int64[site_ids zeros(Int64, n_sites) zeros(Int64, n_sites)]
```

Comment:
Please avoid wall-of-text blocks.

Add spaces between lines if you add comments.

```suggestion
    s_order = Union{Float64,Int64}[Int64.(site_order) rand(n_sites)]

    # All selected sites are in the same reef, so 2 should be replaced
    reef_locs[prefsites] .= "2"

    # Empty ranking just for testing
    rankings = Int64[site_ids zeros(Int64, n_sites) zeros(Int64, n_sites)]
```